ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -544,18 +677,38 @@ private SelectionKey keyForId(String id) {
     return this.keyMap.get(id);
   }
 
+  /**
+   * Handle read/write IOException
+   */
+  private void handleReadWriteIOException(IOException e, SelectionKey key) {
+    String socketDescription = socketDescription(channel(key));
+    if (e instanceof EOFException || e instanceof ConnectException) {
+      metrics.selectorDisconnectedErrorCount.inc();
+      logger.error("Connection {} disconnected", socketDescription, e);
+    } else {
+      metrics.selectorIOErrorCount.inc();
+      logger.warn("Error in I/O with connection to {}", socketDescription, e);
+    }
+  }
+
   /**
    * Process reads from ready sockets
+   * @return the {@link NetworkReceive} if no IOException during read().
    */
-  private void read(SelectionKey key, Transmission transmission) throws IOException {
+  private NetworkReceive read(SelectionKey key, Transmission transmission) throws IOException {
```

Comment:
both `read` and `write` methods can remove `throws IOException` since there is no exception thrown from these two methods.

Reply:
fixed.